### PR TITLE
Add test for APP_BASE_PATH

### PR DESCRIPTION
**Pull request by OGAWAHirofumi**

### CODE REVIEW

File: src/bepasty/tests/test_app.py [hunk 1]
```diff
@@ -24,3 +42,61 @@ def test_secret_key():
...
+        assert url == 'http://localhost' + request.path
+
+    app, client = prepare(none_callback)
+
+    response = client.get('/bepasty/test_call')
+    assert response.status_code == 404
+
+    response = client.get('/test_call')
+    assert response.status_code == 200
+    assert response.data == 'done'.encode()
```
Comment (ThomasWaldmann) [inline src/bepasty/tests/test_app.py:83]: b'done'

File: src/bepasty/tests/test_app.py [hunk 2]
```diff
@@ -24,3 +42,61 @@ def test_secret_key():
...
+        assert url == 'http://localhost' + Config.APP_BASE_PATH + request.path
+
+    app, client = prepare(prefix_callback)
+
+    response = client.get('/test_call')
+    assert response.status_code == 404
+
+    response = client.get('/bepasty/test_call')
+    assert response.status_code == 200
+    assert response.data == 'done'.encode()
```
Comment (ThomasWaldmann) [inline src/bepasty/tests/test_app.py:102]: b'done'

File: src/bepasty/tests/test_app.py [hunk 3]
```diff
@@ -2,11 +2,29 @@
...
+def app_fixture():
+    app_base_path = Config.APP_BASE_PATH
+    permissions = Config.PERMISSIONS
+    secret_key = Config.SECRET_KEY
+
+    yield
+
+    Config.APP_BASE_PATH = app_base_path
+    Config.PERMISSIONS = permissions
+    Config.SECRET_KEY = secret_key
```
Comment (ThomasWaldmann) [inline src/bepasty/tests/test_app.py:24]: did you check if you can use `monkeypatch` instead of this?

File: src/bepasty/tests/test_app.py [hunk 4]
```diff
@@ -24,3 +42,61 @@ def test_secret_key():
...
+
+
+def test_none(app_fixture):
+    Config.APP_BASE_PATH = None
+
+    def none_callback():
+        url = url_for('test.test_call')
+        assert url == request.path
+        url = url_for('test.test_call', _external=True)
+        assert url == 'http://localhost' + request.path
```
Comment (ThomasWaldmann) [inline src/bepasty/tests/test_app.py:74]: hmm, isn't that getting into the "testing flask (not: bepasty)" department?

File: src/bepasty/tests/test_app.py [hunk 5]
```diff
@@ -24,3 +42,61 @@ def test_secret_key():
...
+
+def test_prefix(app_fixture):
+    Config.APP_BASE_PATH = '/bepasty'
+
+    def prefix_callback():
+        url = url_for('test.test_call')
+        assert url == Config.APP_BASE_PATH + request.path
+        url = url_for('test.test_call', _external=True)
+        assert url == 'http://localhost' + Config.APP_BASE_PATH + request.path
+
```
Comment (ThomasWaldmann) [inline src/bepasty/tests/test_app.py:94]: same here? is APP_BASE_PATH a flask or a bepasty feature?

File: src/bepasty/tests/test_app.py (hunk 1 above)
Comment (OGAWAHirofumi) [reply src/bepasty/tests/test_app.py:83]: fixed

File: src/bepasty/tests/test_app.py (hunk 2 above)
Comment (OGAWAHirofumi) [reply src/bepasty/tests/test_app.py:102]: fixed

File: src/bepasty/tests/test_app.py (hunk 3 above)
Comment (OGAWAHirofumi) [reply src/bepasty/tests/test_app.py:24]: looks like works, fixed

File: src/bepasty/tests/test_app.py (hunk 4 above)
Comment (OGAWAHirofumi) [reply src/bepasty/tests/test_app.py:74]: this was from proxy code test. removed _external test

File: src/bepasty/tests/test_app.py (hunk 5 above)
Comment (OGAWAHirofumi) [reply src/bepasty/tests/test_app.py:94]: APP_BASE_PATH is bepasty's PrefixMiddleware() stuff

**Comment by OGAWAHirofumi** [comment]
changes

- use monkeypath
- use b'...'
- remove _external test

File: src/bepasty/tests/test_app.py (hunk 5 above)
Comment (ThomasWaldmann) [reply src/bepasty/tests/test_app.py]: ok

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**